### PR TITLE
Add performance improvement blog

### LINKED ISSUE #2477: OS Performance Improvement Blog
**Describe the blog post your would like to write**
The goal of this blog is to share a systematic look at historical performance initiatives delivered, and provide a forward look at the planned roadmap and investments to come. This will give more broad visibility to the ongoing work, and further engage stakeholders across the community with the progress along the way.

**What is the title of the blog post?**
An update on the OpenSearch Project’s continued performance progress through 2.11

**Who are the authors?**
Who will be listed as the authors of this blog post
Saurabh Singh: @getsaurabh02 
Pallavi Priyadarshini: @Pallavi-AWS
Dagney Braun: @dagneyb 
Rishabh Singh: @rishabh6788 

**What is the proposed posting date?**
First week of Jan

### PR DESCRIPTION
**Pull request by kolchfa-aws**
### Description
Adds the OpenSearch performance improvements blog
 
### Issues Resolved
Closes #2477 

### Check List
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the BSD-3-Clause License.

### CODE REVIEW

File: _posts/2024-01-03-opensearch-performance-improvements.md [hunk 1]
```diff
@@ -0,0 +1,1141 @@
...
+
+* `match_all` queries with sorting showed a more than 20x performance boost across the board because of multiple improvements made in the area (see [#6321](https://github.com/opensearch-project/OpenSearch/pull/6321) and [#7244](https://github.com/opensearch-project/OpenSearch/pull/7244)) and other Lucene enhancements.
+
+* Queries for ascending and descending sort-after-timestamp saw a significant performance improvement of up to 70x overall. The optimizations introduced (such as [#6424](https://github.com/opensearch-project/OpenSearch/pull/6424) and [#8167](https://github.com/opensearch-project/OpenSearch/issues/8167)) extend across various numeric types, including but not limited to `int`, `short`, `float`, `double`, `date`, and others.
+
+* Other popular queries such as `search_after` saw an about 60x reduction in latency, attributed to the improvements made in the area involving optimally skipping segments during search (see [#7453](https://github.com/opensearch-project/OpenSearch/pull/7453)). The `search_after` queries can be used as the recommended alternative to scroll queries for a better search experience.
```
Comment (getsaurabh02) [inline _posts/2024-01-03-opensearch-performance-improvements.md:118]: Can we add a line item after:

* Implementation support for match_only_text field to optimize on storage and indexing/search latency for text queries is in progress ([#11039](https://github.com/opensearch-project/OpenSearch/pull/11039)).

File: _posts/2024-01-03-opensearch-performance-improvements.md [hunk 2]
```diff
@@ -0,0 +1,1141 @@
...
+
+### Time series
+
+In the context of aggregations over range and time-series data, we used the `nyc_taxis` and `http_logs` workloads from OpenSearch Benchmark to benchmark various popular use cases. Here are the key highlights:
+
+* Range queries, popular for aggregation use cases, exhibited about 50%--75% improvement, attributed to system upgrades such as Lucene (from v8.8 in Elasticsearch 7.10.2 to v9.7 in OpenSearch 2.11) and JDK (from JDK15 in Elasticsearch 7.10.2 to JDK17 in OpenSearch 2.11).
+
+* Hourly aggregations and multi-term aggregations also demonstrated improvement, varying from 5% to 35%, attributed to similar time-series improvements discussed previously.
+
+* `date_histograms` and `date_histogram_agg` queries exhibited either comparable or slightly decreased performance, ranging from 5% to around 20% in multi-node environments. These issues are actively being addressed as part of the ongoing project efforts (see [#11083](https://github.com/opensearch-project/OpenSearch/pull/11083)).
```
Comment (getsaurabh02) [inline _posts/2024-01-03-opensearch-performance-improvements.md:129]: Also a line item under Time Series:

* For the date histogram aggregations, there are upcoming changes aiming to improve the performance by rounding-down dates to the nearest interval (such as year, quarter, month, week, day) using SIMD ([#11194](https://github.com/opensearch-project/OpenSearch/pull/11194)).

Comment (kolchfa-aws) [reply _posts/2024-01-03-opensearch-performance-improvements.md:138]: Done

File: _posts/2024-01-03-opensearch-performance-improvements.md [hunk 3]
```diff
@@ -0,0 +1,1152 @@
...
+
+In this blog, we'll share a comprehensive view of strategic enhancements and features in performance that OpenSearch has delivered to date. Additionally, we'll provide a forward look at the [planned roadmap](https://github.com/orgs/opensearch-project/projects/153/views/1) of improvements in open source. We'll compare the core engine performance of the latest OpenSearch version (OpenSearch 2.11) to the state just before the OpenSearch fork, Elasticsearch 7.10.2. We'll highlight continuous advancements made in the OpenSearch core engine, ongoing feature enhancements centered around the popular log analytics and search use cases, and plans to drive improvements for which we are seeking community collaboration.
```
Comment (getsaurabh02) [inline _posts/2024-01-03-opensearch-performance-improvements.md:20]: Can we add below sentence at the end attributing the community:

"These progressions were realized through collaborative efforts with the community, and OpenSearch is actively seeking to enhance community engagement, specifically in the field of improving performance."

File: _posts/2024-01-03-opensearch-performance-improvements.md [hunk 4]
```diff
@@ -0,0 +1,1151 @@
...
+authors:
+  - sisurab
+  - pallp
+  - dagney
+  - rishabhsi
+date: 2024-01-03
+categories:
+    - technical-posts
+    - community
+meta_keywords:
```
Comment (pajuric) [inline _posts/2024-01-03-opensearch-performance-improvements.md:13]: Please add the following meta: 

**meta_keywords:**  OpenSearch performance improvements, OpenSearch roadmap, high volume indexing, low latency search
**meta_description:** Learn more about the OpenSearch Project roadmap and how the project improved the performance of its core open source engine to drive down latency and improve efficiency.

File: _posts/2024-01-03-opensearch-performance-improvements.md [hunk 5]
```diff
@@ -0,0 +1,1151 @@
...
+The following sections present OpenSearch 2.11 features that improve performance.
+
+### LogByteSize merge policy
+
+In the realm of log analytics, the Tiered merge policy has been a cornerstone for efficient shard merges. In OpenSearch 2.11 we introduced the LogByteSize merge policy. This new approach consistently merges adjacent segments, proving especially advantageous for time-series data characterized by timestamp sorting and minimal timestamp overlap between segments.
+
+The following are the key findings from this exercise.
+
+* Timestamp queries with ascending sort had over 75% improvement. This transformation is attributable to the impactful contribution of enhancement [#9241](https://github.com/opensearch-project/OpenSearch/issues/9241).
+* About 40% enhancement in descending sort timestamp queries, surpassing tiered merge policy.
```
Comment (natebower) [inline _posts/2024-01-03-opensearch-performance-improvements.md:699]:
```suggestion
* About a 40% enhancement in descending sort timestamp queries, surpassing the Tiered merge policy.
```

File: _posts/2024-01-03-opensearch-performance-improvements.md [hunk 6]
```diff
@@ -0,0 +1,1151 @@
...
+* Use cases around ascending and descending sort with_after_timestamp saw regression which is a known case for smaller workload with this merge policy.
```
Comment (natebower) [inline _posts/2024-01-03-opensearch-performance-improvements.md:700]:
```suggestion
* Use cases around ascending and descending sort with _after_ timestamp saw regression, which is a known case for smaller workloads with this merge policy.
```

Comment (natebower) [inline _posts/2024-01-03-opensearch-performance-improvements.md:700]: This reads as though either "a" or "the" should precede _after_.

File: _posts/2024-01-03-opensearch-performance-improvements.md [hunk 7]
```diff
@@ -0,0 +1,1151 @@
...
+* Other common use cases for log analytics, such as Multi-term aggregation, hourly_agg, range, and scroll queries exhibited comparable performance, with a subtle improvement of less than 5% attributed to the new segment merge policy.
```
Comment (natebower) [inline _posts/2024-01-03-opensearch-performance-improvements.md:701]:
```suggestion
* Other common use cases for log analytics, such as multi-term aggregation, hourly_agg, range, and scroll queries exhibited comparable performance, with a subtle improvement of less than 5% attributed to the new segment merge policy.
```

File: _posts/2024-01-03-opensearch-performance-improvements.md [hunk 8]
```diff
@@ -0,0 +1,1151 @@
...
+
+#### Detailed results: OS 2.11 Tiered merge policy vs OS 2.11 LogByteSize merge policy
+
+**Setup**: 3 data nodes (r5.xlarge) with 32 GB RAM and 16 GB heap. 3 cluster manager nodes (c5.xlarge) with 8 GB RAM and 4 GB heap. Index settings: 3 shards and 0 replicas, `max_segment_size`: 500 MB, `refresh_interval`: 1 s.
+
+**`http_logs` workload results:** The following table illustrates a benchmark comparison for the `http_logs` workload for OpenSearch 2.11 with Tiered merge policy vs. LogByteSize merge policy. It includes the 90th percentile of `took` time latency measurements for each (p90) and the observed percentage improvements.
```
Comment (natebower) [inline _posts/2024-01-03-opensearch-performance-improvements.md:707]:
```suggestion
**`http_logs` workload results**: The following table provides a benchmark comparison for the `http_logs` workload for OpenSearch 2.11 with the Tiered merge policy vs. the LogByteSize merge policy. It includes the 90th percentile of `took` time latency measurements for each (p90) and the observed percentage improvements.
```

File: _posts/2024-01-03-opensearch-performance-improvements.md [hunk 9]
```diff
@@ -0,0 +1,1151 @@
...
+        <td>6</td>
+        <td>14%</td>
+    </tr>
+</table>
+
+### Zsandard codec compression
+
+This addition empowers OpenSearch users with the new Zstandard compression codecs for their data. Users can specify `zstd` or `zstd_no_dict` in the `index.codec` setting during index creation or [modify the codecs for existing indexes](https://opensearch.org/docs/latest/im-plugin/index-codecs/#changing-an-index-codec). OpenSearch will continue to support the existing `zlib` and `lz4` codecs, with the default as `lz4`.
+
+The following sections contain benchmarking results, representing the average over 5 days of data from nightly runs using OpenSearch Benchmark.
```
Comment (natebower) [inline _posts/2024-01-03-opensearch-performance-improvements.md:806]:
```suggestion
The following sections contain benchmarking results representing the average of 5 days of data from nightly runs using OpenSearch Benchmark.
```

File: _posts/2024-01-03-opensearch-performance-improvements.md [hunk 10]
```diff
@@ -0,0 +1,1151 @@
...
+
+#### Highlights
+
+Here are the key highlights:
+
+* A notable increase in the ingestion throughput, ranging from 5% to 8%, attributed to the new `zstd` codec. This enhancement owes its success to codec-related pull requests ([#7908](https://github.com/opensearch-project/OpenSearch/pull/7908), [#7805](https://github.com/opensearch-project/OpenSearch/issues/7805), and [#7555](https://github.com/opensearch-project/OpenSearch/issues/7555)).
+
+* About 30% reduction in on-disk data size, surpassing the default `lz4` codec for unparalleled efficiency, all while maintaining a near-identical CPU utilization pattern compared to the default compression.
```
Comment (natebower) [inline _posts/2024-01-03-opensearch-performance-improvements.md:814]:
```suggestion
* About a 30% reduction in on-disk data size, surpassing the default `lz4` codec for unparalleled efficiency, all while maintaining a near-identical CPU utilization pattern compared to the default compression.
```

File: _posts/2024-01-03-opensearch-performance-improvements.md [hunk 11]
```diff
@@ -0,0 +1,1151 @@
...
+
+* The search p90 latencies remained virtually unchanged, with negligible differences of less than 2% in a few areas.
+
+#### Detailed results: OpenSearch 2.11 default (`lz4`) compression vs. OpenSearch 2.11 `zstd` compression using multiple shards
+
+**Setup:** 3 data nodes (r5.xlarge) with 32 GB RAM and 16 GB heap. 3 cluster manager nodes (c5.xlarge) with 8 GB RAM and 4G Heap. Index settings: 3 shards and 0 replicas.
```
Comment (natebower) [inline _posts/2024-01-03-opensearch-performance-improvements.md:820]:
```suggestion
**Setup**: 3 data nodes (r5.xlarge) with 32 GB RAM and 16 GB heap. 3 cluster manager nodes (c5.xlarge) with 8 GB RAM and 4 G heap. Index settings: 3 shards and 0 replicas.
```

File: _posts/2024-01-03-opensearch-performance-improvements.md [hunk 12]
```diff
@@ -0,0 +1,1151 @@
...
+
+**Indexing throughput results (docs/sec)**: The following table presents the indexing throughput comparison of the `http_logs` and `nyc_taxis` workloads for OpenSearch 2.11 with default vs. `zstd` codec enabled, including its percentage improvement.
```
Comment (natebower) [inline _posts/2024-01-03-opensearch-performance-improvements.md:822]:
```suggestion
**Indexing throughput results (docs/sec)**: The following table provides an indexing throughput comparison of the `http_logs` and `nyc_taxis` workloads for OpenSearch 2.11 with the default codec vs. the `zstd` codec enabled and includes the percentage improvement observed when using `zstd`.
```

Comment (natebower) [inline _posts/2024-01-03-opensearch-performance-improvements.md:822]: Confirm that my changes retain the intended meaning.

File: _posts/2024-01-03-opensearch-performance-improvements.md [hunk 13]
```diff
@@ -0,0 +1,1151 @@
...
+    </tr>
+    <tr class="light-green-clr">
+        <td><code>nyc_taxis</code></td>
+        <td>118,123.5</td>
+        <td>127,131</td>
+        <td>8%</td>
+    </tr>
+</table>
+
+**`nyc_taxis` search workload results**: The following table illustrates a benchmark comparison for the `nyc_taxis` workload for OpenSearch 2.11 with default codec vs. `zstd` codec enabled, including percentage improvement.
```
Comment (natebower) [inline _posts/2024-01-03-opensearch-performance-improvements.md:845]:
```suggestion
**`nyc_taxis` search workload results**: The following table provides a benchmark comparison for the `nyc_taxis` workload for OpenSearch 2.11 with the default codec vs. the `zstd` codec enabled and includes the percentage improvement observed when using `zstd`.
```

Comment (natebower) [inline _posts/2024-01-03-opensearch-performance-improvements.md:845]: Confirm that my changes retain the intended meaning.

File: _posts/2024-01-03-opensearch-performance-improvements.md [hunk 14]
```diff
@@ -0,0 +1,1151 @@
...
+        <td><code>range</code></td>
+        <td>74.5</td>
+        <td>77.5</td>
+        <td>-4%</td>
+    </tr>
+</table>
+
+#### Detailed results: Index data size on disk (bytes) with `zstd` compression using a single shard
+
+**Setup:** OpenSearch 2.11.0, single node (r5.xlarge) with 32 GB RAM and 16 GB heap. Index settings: 1 shard and 0 replicas.
```
Comment (natebower) [inline _posts/2024-01-03-opensearch-performance-improvements.md:888]:
```suggestion
**Setup**: OpenSearch 2.11.0, single node (r5.xlarge) with 32 GB RAM and 16 GB heap. Index settings: 1 shard and 0 replicas.
```

File: _posts/2024-01-03-opensearch-performance-improvements.md [hunk 15]
```diff
@@ -0,0 +1,1151 @@
...
+
+**Data size of disk results (bytes)**: The following table illustrates a benchmark comparison of the on-disk data size for the `http_logs` and `pmc` workloads for OpenSearch 2.11 with default vs. `zstd` codec enabled, including percentage improvement.
```
Comment (natebower) [inline _posts/2024-01-03-opensearch-performance-improvements.md:890]:
```suggestion
**Data size of disk results (bytes)**: The following table provides a benchmark comparison of the on-disk data size for the `http_logs` and `pmc` workloads for OpenSearch 2.11 with the default codec vs. the `zstd` codec enabled and includes the percentage improvement observed when using `zstd`.
```

Comment (natebower) [inline _posts/2024-01-03-opensearch-performance-improvements.md:890]: Confirm that my changes retain the intended meaning.

File: _posts/2024-01-03-opensearch-performance-improvements.md [hunk 16]
```diff
@@ -0,0 +1,1151 @@
...
+        <td>15,608,881,718.5</td>
+        <td>15,822,040,185</td>
+        <td>25%</td>
+        <td>24%</td>
+    </tr>
+</table>
+
+### Concurrent search improvements (*Experimental in 2.11*)
+
+OpenSearch users can now achieve better execution speed with concurrent segment search launched as experimental in OpenSearch 2.11. By default, OpenSearch processes a request sequentially across all the data segments on each shard during the query phase of a search request execution. With concurrent search, every shard-level request can concurrently search across segments during the query phase. Each shard divides its segments into [multiple slices](https://opensearch.org/blog/concurrent_segment_search/), where each slice serves as a unit of work executed in parallel on a separate thread. Therefore, the slice count governs the maximum degree of parallelism for a shard-level request. After all the slices finish their tasks, OpenSearch executes a reduce operation on the slices, merging them to generate the final result for the shard-level request.
```
Comment (natebower) [inline _posts/2024-01-03-opensearch-performance-improvements.md:921]:
```suggestion
OpenSearch users can now achieve better execution speed with concurrent segment search, launched as experimental in OpenSearch 2.11. By default, OpenSearch processes a request sequentially across all the data segments on each shard during the query phase of a search request execution. With concurrent search, every shard-level request can concurrently search across segments during the query phase. Each shard divides its segments into [multiple slices](https://opensearch.org/blog/concurrent_segment_search/), where each slice serves as a unit of work executed in parallel on a separate thread. Therefore, the slice count governs the maximum degree of parallelism for a shard-level request. After all the slices finish their tasks, OpenSearch executes a reduce operation on the slices, merging them to generate the final result for the shard-level request.
```

File: _posts/2024-01-03-opensearch-performance-improvements.md (hunk 9 above)
Comment (natebower) [inline _posts/2024-01-03-opensearch-performance-improvements.md:806]: "of over 5 days"?

File: _posts/2024-01-03-opensearch-performance-improvements.md [hunk 17]
```diff
@@ -0,0 +1,1151 @@
...
+
+The benchmark results presented below show the benefits of concurrent search in action. These are the averages of data generated from over 4 days of nightly runs using OpenSearch Benchmark.
+
+#### Highlights
+
+Here are the key highlights:
+
+* An increase in performance with aggregate queries on workloads such as `nyc_taxis` workload, showcasing an improvement ranging between 50% to 70% over the default configuration.
```
Comment (natebower) [inline _posts/2024-01-03-opensearch-performance-improvements.md:929]:
```suggestion
* An increase in performance with aggregate queries on workloads such as the `nyc_taxis` workload, showcasing an improvement ranging between 50% and 70% over the default configuration.
```

File: _posts/2024-01-03-opensearch-performance-improvements.md [hunk 18]
```diff
@@ -0,0 +1,1151 @@
...
+* The log analytics use cases for range queries demonstrated an improvement of around 65%.
+* Aggregation queries with hourly data aggregations, such as those for the `http_logs` `hourly_agg` workload, demonstrated a boost of up to 50% in performance.
+* Comparable latencies for auto or date histogram queries, with no noteworthy improvement or regression in performance.
+* `multi_term_agg`, `asc_sort_size`, `dec_sort_size` and `scroll` queries showed regression. To delve deeper into the intricacies, the concurrent search contributors are proactively addressing this in the upcoming OpenSearch 2.12 GA release.
```
Comment (natebower) [inline _posts/2024-01-03-opensearch-performance-improvements.md:933]:
```suggestion
* `multi_term_agg`, `asc_sort_size`, `dec_sort_size`, and `scroll` queries showed regression. To delve deeper into the intricacies, the concurrent search contributors are proactively addressing this in the upcoming OpenSearch 2.12 GA release.
```

File: _posts/2024-01-03-opensearch-performance-improvements.md [hunk 19]
```diff
@@ -0,0 +1,1151 @@
...
+
+#### Detailed results: OpenSearch 2.11 with concurrent search enabled vs. disabled
+
+**Setup**: OpenSearch 2.11.0 single node (r5.2xlarge) with 64 GB RAM and 32 GB heap. Index settings: 1 shard and 0 replicas.
+
+**`nyc_taxis` workload results:** The following table illustrates a benchmark comparison of the `nyc_taxis` workload for OpenSearch 2.11 with concurrent search disabled and enabled (with 0 slices and with 4 slices). It includes the 90th percentile of `took` time latency measurements for each (p90) and the observed percentage improvements.
```
Comment (natebower) [inline _posts/2024-01-03-opensearch-performance-improvements.md:939]:
```suggestion
**`nyc_taxis` workload results**: The following table provides a benchmark comparison of the `nyc_taxis` workload for OpenSearch 2.11 with concurrent search disabled and enabled (with 0 slices and with 4 slices). It includes the 90th percentile of `took` time latency measurements for each (p90) and the observed percentage improvements.
```

File: _posts/2024-01-03-opensearch-performance-improvements.md [hunk 20]
```diff
@@ -0,0 +1,1151 @@
...
+        <td><code>range</code></td>
+        <td>201</td>
+        <td>73</td>
+        <td>77</td>
+        <td>64%</td>
+        <td>62%</td>
+    </tr>
+</table>
+
+**`http_logs` workload results:** The following table illustrates a benchmark comparison of the `http_logs` workload for OpenSearch 2.11 with concurrent search disabled and enabled (with 0 slices and with 4 slices). It includes the 90th percentile of latency measurements for each (p90) and the observed percentage improvements.
```
Comment (natebower) [inline _posts/2024-01-03-opensearch-performance-improvements.md:992]:
```suggestion
**`http_logs` workload results**: The following table provides a benchmark comparison of the `http_logs` workload for OpenSearch 2.11 with concurrent search disabled and enabled (with 0 slices and with 4 slices). It includes the 90th percentile of latency measurements for each (p90) and the observed percentage improvements.
```

File: _posts/2024-01-03-opensearch-performance-improvements.md [hunk 21]
```diff
@@ -0,0 +1,1151 @@
...
+        <td>3</td>
+        <td>3</td>
+        <td>0%</td>
+        <td>0%</td>
+    </tr>
+</table>
+
+* * *
+
+*We would like to take this opportunity to thank the OpenSearch core developers for their contributions to the technical roadmap. We sincerely appreciate all the suggestions from Michael Froh, Andriy Redko, Jonah Kowall, Amitai Stern, Jon Handler, Prabhakar Sithanandam, Mike McCandles, Anandhi Bumstead, Eli Fisher, Carl Meadows, and Mukul Karnik towards writing this blog. Credits to Fanit Kolchina and Nathan Bower for editing and Carlos Canas for creating the graphics.*
```
Comment (natebower) [inline _posts/2024-01-03-opensearch-performance-improvements.md:1151]:
```suggestion
*We would like to take this opportunity to thank the OpenSearch core developers for their contributions to the technical roadmap. We sincerely appreciate all the suggestions from Michael Froh, Andriy Redko, Jonah Kowall, Amitai Stern, Jon Handler, Prabhakar Sithanandam, Mike McCandles, Anandhi Bumstead, Eli Fisher, Carl Meadows, and Mukul Karnik in writing this blog post. Credits to Fanit Kolchina and Nathan Bower for editing and Carlos Canas for creating the graphics.*
```

Comment (natebower) [inline _posts/2024-01-03-opensearch-performance-improvements.md:1151]: Confirm that "McCandles" shouldn't be "McCandless".

File: _posts/2024-01-03-opensearch-performance-improvements.md [hunk 22]
```diff
@@ -0,0 +1,1151 @@
...
+
+The benchmark results presented below show the benefits of concurrent search in action. These are the averages of data generated from over 4 days of nightly runs using OpenSearch Benchmark.
+
+#### Highlights
+
+Here are the key highlights:
+
+* An increase in performance with aggregate queries on workloads such as `nyc_taxis` workload, showcasing an improvement ranging between 50% to 70% over the default configuration.
+* The log analytics use cases for range queries demonstrated an improvement of around 65%.
+* Aggregation queries with hourly data aggregations, such as those for the `http_logs` `hourly_agg` workload, demonstrated a boost of up to 50% in performance.
```
Comment (natebower) [inline _posts/2024-01-03-opensearch-performance-improvements.md:931]: "such as those for the `hourly_agg` operation on the `http_logs` workload" (`hourly_agg` is not a workload)?

File: _authors/sisurab.markdown [hunk 23]
```diff
@@ -6,4 +6,4 @@ github: getsaurabh02
 linkedin: getsaurabh02
 ---
 
-**Saurabh** is a Senior Software Engineer working on OpenSearch at Amazon Web Services. He is passionate about solving problems in the large-scale distributed systems. He is an active contributor to OpenSearch.
+**Saurabh Singh** is an Engineering Lead working on OpenSearch at Amazon Web Services, leading the core search performance space. He is passionate about solving problems in the large-scale distributed systems. He is an active OpenSearch contributor.
```
[the remark this answers is not in the document]
Comment (kolchfa-aws) [reply _authors/sisurab.markdown:9]: @getsaurabh02 What do you think?

File: _posts/2024-01-03-opensearch-performance-improvements.md (hunk 12 above)
Comment (kolchfa-aws) [reply _posts/2024-01-03-opensearch-performance-improvements.md:822]: @getsaurabh02 Could you confirm that this suggestion is accurate?

File: _posts/2024-01-03-opensearch-performance-improvements.md (hunk 22 above)
Comment (kolchfa-aws) [reply _posts/2024-01-03-opensearch-performance-improvements.md:931]: @getsaurabh02 Could you confirm that we can make this change?

File: _posts/2024-01-03-opensearch-performance-improvements.md [hunk 24]
```diff
@@ -0,0 +1,1151 @@
...
+
+In version 2.10, OpenSearch introduced **_remote-backed storage_**, allowing users to directly write these segments to object storage, such as Amazon-managed S3 or Oracle Blob Storage, to improve data durability. With remote-backed storage, in addition to storing data on a local disk, all the ingested data is stored in the configured remote store. At every refresh, the remote store also automatically becomes a point-in-time recovery point, helping users achieve a recovery point objective (RPO) of zero with the same durability properties of the configured remote store. To learn more, see the OpenSearch [remote-backed storage blog](https://opensearch.org/blog/remote-backed-storage/).
```
[the remark this answers is not in the document]
Comment (kolchfa-aws) [reply _posts/2024-01-03-opensearch-performance-improvements.md]: @getsaurabh02 Is it Oracle Cloud Infrastructure (OCI) Object Storage or Azure Blob Storage?

File: _posts/2024-01-03-opensearch-performance-improvements.md [hunk 25]
```diff
@@ -0,0 +1,1151 @@
...
+
+* Other popular queries such as `search_after` saw an about 60x reduction in latency, attributed to the improvements made in the area involving optimally skipping segments during search (see [#7453](https://github.com/opensearch-project/OpenSearch/pull/7453)). The `search_after` queries can be used as the recommended alternative to scroll queries for a better search experience.
+
+* Implementation support for `match_only_text` field to optimize on storage and indexing/search latency for text queries is in progress (see [#11039](https://github.com/opensearch-project/OpenSearch/pull/11039)).
+
+### Time series
+
+In the context of aggregations over range and time-series data, we used the `nyc_taxis` and `http_logs` workloads from OpenSearch Benchmark to benchmark various popular use cases. Here are the key highlights:
```
[the remark this answers is not in the document]
Comment (getsaurabh02) [reply _posts/2024-01-03-opensearch-performance-improvements.md:131]: These refer to queries where users aggregate over date ranges, so I would keep it as "over"

File: _posts/2024-01-03-opensearch-performance-improvements.md (hunk 12 above)
Comment (getsaurabh02) [reply _posts/2024-01-03-opensearch-performance-improvements.md:822]: looks good to me

File: _posts/2024-01-03-opensearch-performance-improvements.md (hunk 6 above)
Comment (kolchfa-aws) [reply _posts/2024-01-03-opensearch-performance-improvements.md]: Changed to "workloads"

File: _posts/2024-01-03-opensearch-performance-improvements.md (hunk 22 above)
Comment (getsaurabh02) [reply _posts/2024-01-03-opensearch-performance-improvements.md:931]: looks good to me, OR "such as those for the `http_logs` workload"
yes - hourly_agg is not a workload)

File: _authors/sisurab.markdown (hunk 23 above)
[the remark this answers is not in the document]
Comment (getsaurabh02) [reply _authors/sisurab.markdown]: I am fine with either.

**Comment by kolchfa-aws** [comment]
@pajuric The blog is ready to publish. Thanks!

**Comment by pajuric** [comment]
@nateynateynate @dtaivpp @krisfreedain - This blog is ready to push live. If possible, can we get this out by 12PM PST today, please.

**Review by nateynateynate — APPROVED** [review]
Looks good to me! Great job!

File: _posts/2024-01-03-opensearch-performance-improvements.md [hunk 26]
```diff
@@ -0,0 +1,1151 @@
...
+
+This concludes the main summary of OpenSearch performance improvements to date. The following appendix sections provide the benchmarking details for readers interested in replicating any run.
+
+* * *
+
+## Appendix: Detailed execution and results
+
+If you're interested in the details of the performance benchmarks we used, exploring the methodologies behind their execution, or examining the comprehensive results, keep reading. For OpenSearch users interested in establishing benchmarks and replicating these runs, we've provided comprehensive setup details alongside each result. This section provides the core engine performance comparison between the latest OpenSearch version (OpenSearch 2.11) and the state just before the OpenSearch fork, Elasticsearch 7.10.2, with a mid-point performance measurement on OpenSearch 2.3.
```
Comment (getsaurabh02) [inline _posts/2024-01-03-opensearch-performance-improvements.md:172]: Can we add a line at the end of this paragraph:

"Also, we've identified items in the [performance roadmap](https://github.com/orgs/opensearch-project/projects/153/views/1) that require active enhancements due to observed regressions in specific areas."

Comment (getsaurabh02) [reply _posts/2024-01-03-opensearch-performance-improvements.md:172]: ohh looks like its merged! 👍

**Pull request merged**